### ISSUE TITLE
HTTP Method 'GET' not allowed.

### ISSUE THREAD

**Issue by fenixjke** [xjsender/haoide#128]
Hi,
I got an error when I tried to run sync test (command + shift + U):
errorCode: METHOD_NOT_ALLOWED            	
Time Stamp: 2016-06-21 17:36:30           	
message: HTTP Method 'GET' not allowed. Allowed are POST

**Comment by xjsender** [comment]
@fenixjke , how to replicate it?

**Comment by superdave7380** [comment]
Hey, I also get the error whenever I run to run a sync test.
When on test class, right-click > Haoide > Run Sync Test
Happens every time.

Thanks!

**Comment by xjsender** [comment]
@superdave7380 , yes, I also replicate this issue caused by new feature of v37.0, you can decrease the api version to avoid this problem.

**Comment by superdave7380** [comment]
As always, thanks very much for the speedy reply.

**Comment by aitiancai** [comment]
change api_version to 35 in Setting-user。

**Comment by rspurrell** [comment]
I would like this issue addressed as well for all of the same reasons that @scolladon mentioned.

**Comment by adashko** [comment]
> I would like this issue addressed as well for all of the same reasons that @scolladon mentioned.

Sync test working (after your fix), but 'View Code Coverage' - not. 'No code coverage cache, please execute `Run Sync Test` on related test class before view code coverage'

**Comment by rspurrell** [comment]
> > I would like this issue addressed as well for all of the same reasons that @scolladon mentioned.
> 
> Sync test working (after your fix), but 'View Code Coverage' - not. 'No code coverage cache, please execute `Run Sync Test` on related test class before view code coverage'

Since I pushed this fix it seems that the lines not covered are no longer being returned in the code coverage as part of the RunTestsSynchronous response.  The locationsNotCovered array now contains empty entries.  This is possibly a SF bug, but it may be intentional in favor of another method.  Either way, the response no longer returns what HaoIDE is expecting.

After inspecting with Fiddler, I can see that this is happening in the Developer Console (DC) too.  However, when viewing code coverage in the DC, a separate query for coverage is made which does return the lines covered and uncovered.  Unfortunately, the response from that call is very different from what is currently handled.

If this isn't a SF bug, two updates will need to be made. One in the processor.handle_run_sync_test function to retrieve the coverage using a query, as in the DC.  The other would be in the util.view_coverage function to correctly read the new JSON structure of said query.

I'll submit a SF bug report tomorrow and see what they say.  However, I don't expect a speedy response.  Likely they'll instruct me to use the code coverage query method.

I'll keep you posted.
